src/app/multi-upload/multi-upload.component.ts

```diff
@@ -0,0 +1,63 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+
+@Component({
+  selector: 'ru-multi-upload',
+  styleUrls: ['multi-upload.component.scss'],
+  templateUrl: 'multi-upload.component.html'
+})
+export class MultiUploadComponent {
+  @Input() acceptedFiles: string = 'image/*';
+  @Input() maxFileSize: number; //MB
+  @Output() onFileAdd = new EventEmitter<File[]>();
+  @Output() onFileChange = new EventEmitter<File[]>();
+  files?: File[];
+  private megabyte: 1000000;
+
+  click(event: Event) {
+    this.addFileFromInputElement(<HTMLInputElement> event.srcElement);
+  }
+
+  drop(event: Event) {
+    this.preventDefault(event);
+    this.addFileFromDragEvent(<DragEvent> event);
+  }
+
+  preventDefault(event: Event) {
+    event.preventDefault();
+    event.stopPropagation();
+  }
+
+  private addFileFromInputElement(uploadInput: HTMLInputElement) {
+    const files = uploadInput.files;
+    if (!files) { return; }
+    if (files.length === 0) { return; }
+    this.addFiles(Array.prototype.slice.call(files));
```

Comment:
`Array.prototype.slice.call(files)` is not very readable. i would prefer another way for this.